api-consumption/src/main/java/org/zalando/nakadi/service/subscription/StreamingContext.java

```diff
@@ -277,10 +280,17 @@ public boolean isSubscriptionConsumptionBlocked() {
     }
 
     public boolean isConsumptionBlocked(final ConsumedEvent event) {
+        final var metadataEventTypeName = kafkaRecordDeserializer.getEventTypeName(event.getEvent());
```

Comment:
would it be possible to utilise parser from BatchItem to parse only event_type from metadata?

Reply:
BatchItem unfortunately does not offer such functionality, it either parsers strictly or non strictly from what I see.

Reply:
Since org.json lib didnt have such feature, I have used Jackson instead. Will only parse metadata